src/main/java/io/antmedia/statistic/StatsCollector.java

```diff
@@ -588,28 +588,24 @@ public boolean enoughResource(){
 		if(cpuLoad < cpuLimit) 
 		{
 			long freeJvmRamValue = getFreeRam();
-			
-			if (freeJvmRamValue > minFreeRamSize) 
-			{
-				long maxPhysicalBytes = Pointer.maxPhysicalBytes();
-				long physicalBytes = Pointer.physicalBytes();
-				if (maxPhysicalBytes > 0) 
-				{
-					long freeNativeMemory = SystemUtils.convertByteSize(maxPhysicalBytes - physicalBytes, "MB"); 
-					if (freeNativeMemory > minFreeRamSize )
-					{
+
+			if (freeJvmRamValue > minFreeRamSize) {
+				long maxPhysicalBytes = SystemUtils.convertByteSize(Pointer.maxPhysicalBytes(), "MB"); 
+				long physicalBytes = SystemUtils.convertByteSize(Pointer.physicalBytes(), "MB"); 
+
+				if ((maxPhysicalBytes-physicalBytes) > minFreeRamSize ){
+					long freeNativeMemory = SystemUtils.convertByteSize(Pointer.availablePhysicalBytes(), "MB"); 										
+
+					if(freeNativeMemory > minFreeRamSize) {
```

Comment:
Why are we checking this value? It's about the system memory.

Reply:
In some cases, native system memory may not be enough for resources.

Reply:
I think there is a kind of misunderstanding about the concept. 

Let me explain with a sample. System has 12GB total memory and 5 GB free memory.  JVM may max 2GB memory.   After JVM consume all of this 2GB memory, system still has 5GB free memory.  However when JVM using 2GB memory, it cannot allocate more memory in the system even if there is 5GB memory. So checking system memory when there is no resource for JVM create false results.   

Does it make sense? 

Please let me know if I'm wrong.

Reply:
Yes, you're right. I was a little confused. Now, I understood completely.

Reply:
Btw, I think minFreeRamSize value should be higher than 20 MB. Because Streaming consume values(average) is in below:

Adaptive Streaming enabled with WebRTC Frame Rate: 20 and Resolution values are 1080,720,480,360,240 resolution default values using:

WebRTC Publish consumes 80-100 MB.
RTMP Publish consumes 75-80 MB.

Adaptive Streaming disabled values are in below:

WebRTC Publish consumes 5-10 MB.
RTMP Publish consumes 5-10 MB.

Reply:
When 1080,720,480,360,240  adaptive settings are enabled, WebRTC publish consumes 80-100MB. Is it right?

What do you recommend for the minFreeRamSize?

Reply:
> When 1080,720,480,360,240 adaptive settings are enabled, WebRTC publish consumes 80-100MB. Is it right?

Yes, alright. Also, it's increasing to 200 MB.

I recommended at least 100MB.

Reply:
Let's make it 50MB. 
Because generally 5 different bitrates (1080, 720, 480, 360,240) are not used at the same time. Having 5 different bitrates may seem to be waste of resources as well.  Mostly, two different bitrates are used.